src/kernel/task/task_resource.c

```diff
@@ -98,11 +100,16 @@ static int task_resource_module_init(void) {
 		assert(res->resource_offset);
 		*res->resource_offset = offset;
 
-		assert((offset + res->resource_size) <= TASK_RESOURCE_SIZE);
+		//assert((offset + res->resource_size) <= TASK_RESOURCE_SIZE);
+
 		offset += binalign_bound(res->resource_size, sizeof(void *));
+		/* if we have aligned resource we have to reserve space for alignment*/
+		if(res->resource_align) {
+			offset += binalign_bound(res->resource_align, sizeof(void *));
+		}
 	}
 
-	assert(offset == binalign_bound(TASK_RESOURCE_SIZE, sizeof(void *)));
+	//assert(offset == binalign_bound(TASK_RESOURCE_SIZE, sizeof(void *)));
```

Comment:
Of couse it's mandatory, but if we have alignment this assertion rises
May be change the condition to '=>'?